src/payload.jl

```diff
@@ -2,14 +2,29 @@
 # ParsedPayload
 #
 
-# What we send to the consume! method
+"""
+    ParsedPayload{B, C<:AbstractParsingContext}
+
+A payload of parsed results, which is passed to `consume!` after each `populate_result_buffer!` call.
+
+# Fields:
+- `row_num::Int`: row number of the first row in the payload
+- `len::Int`: number of rows in the payload
+- `results::B`: parsed result buffer
```

Comment:
what type is `B`? is it `B <:AbstractResultBuffer` where `B` equals the type passed to `parse_file_parallel`?

Reply:
Yeah. I wonder if we really need all result buffers to be `<:AbstractResultBuffer` though. This means that we can't use simple `Vectors` which is a bit sad... ChunkedCSV and ChunkedJSONL do have to define very specific buffer types, so I thought this abstraction has its place, but those buffer types only define an `empty!` and `ensureroom`methods,  populating those is not really abstracted, so I think in time we may want to lift this restriction.